embulk-standards/src/test/java/org/embulk/standards/TestJsonFormatterPlugin.java

```diff
@@ -0,0 +1,141 @@
+import static org.embulk.spi.type.Types.JSON;
+import static org.embulk.spi.type.Types.LONG;
+import static org.embulk.spi.type.Types.STRING;
+import static org.embulk.spi.type.Types.TIMESTAMP;
+import org.embulk.spi.util.Newline;
+import org.embulk.spi.util.Pages;
+import org.joda.time.DateTimeZone;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.msgpack.value.ImmutableMapValue;
+import static org.msgpack.value.ValueFactory.newBoolean;
+import static org.msgpack.value.ValueFactory.newInteger;
+import static org.msgpack.value.ValueFactory.newMap;
+import static org.msgpack.value.ValueFactory.newString;
+import java.nio.charset.Charset;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+
+public class TestJsonFormatterPlugin
+{
+    @Rule
+    public EmbulkTestRuntime runtime = new EmbulkTestRuntime();
+
+    private JsonFormatterPlugin plugin;
+    private MockFileOutput output;
+
+    @Before
+    public void createResource()
+    {
+        plugin = new JsonFormatterPlugin();
+        output = new MockFileOutput();
+    }
+
+    @Test
+    public void checkDefaultConfig()
+    {
+        ConfigSource config = Exec.newConfigSource();
+
+        JsonFormatterPlugin.PluginTask task = config.loadConfig(JsonFormatterPlugin.PluginTask.class);
+        assertEquals(Charset.forName("utf-8"), task.getCharset());
+        assertEquals(Newline.CRLF, task.getNewline());
+        assertEquals(DateTimeZone.UTC, task.getDefaultTimeZone());
+        assertEquals("%Y-%m-%d %H:%M:%S.%6N %z", task.getDefaultTimestampFormat());
+    }
+
+    @Test
+    public void checkOptionalConfig()
+    {
+        ConfigSource config = Exec.newConfigSource()
+                .set("charset", "utf-16")
+                .set("newline", "LF");
+
+        JsonFormatterPlugin.PluginTask task = config.loadConfig(JsonFormatterPlugin.PluginTask.class);
+        assertEquals(Charset.forName("utf-16"), task.getCharset());
+        assertEquals(Newline.LF, task.getNewline());
+    }
+
+    @Test
+    public void testEncodeData()
+    {
+        ConfigSource config = Exec.newConfigSource()
+                .set("charset", "utf-16")
+                .set("newline", "LF");
+        final Schema schema = getSampleSchema();
+
+        plugin.transaction(config, schema, new FormatterPlugin.Control() {
+            @Override
+            public void run(TaskSource taskSource)
+            {
+                PageOutput pageOutput = plugin.open(taskSource, schema, output);
+                List<Page> pages = buildPage(schema,
+                        123.45, "embulk", 1L, true, Timestamp.ofEpochSecond(1422386629), getSampleJson(),
+                        null, null, null, null, null, null
+                );
+                for (Page page : pages) {
+                    pageOutput.add(page);
+                }
+                pageOutput.finish();
+                List<Object[]> records =  Pages.toObjects(schema, pages);
```

Comment:
Build PageBuilder and read from PageReader.
But It's better to compare against formatted JSON string.
```
assertEquals("{123.45, \"embulk\", 1, true...}", val)
```

So I'm thinking about to add MockLineEncoder as @muga  is saying [at above](https://github.com/embulk/embulk/pull/400#issuecomment-187068463)